src/fitnesseMain/badCodeToTestCheckstyle.java

```diff
@@ -0,0 +1,56 @@
+package fitnesseMain;
+
+import java.util.*; //BAD: star imports
+
+class  Hello
+{
+  public final static String hello = "helloWorld!"; //BAD: modifier order
+
+  public String badCommaWhiteSpaceBeforeAndAfter(String input ,String anotherInput)
+    {
+        return "badCommaWhiteSpaceBeforeAndAfter";
+    }
+
+    public void badGenericsWhiteSpace()
+    {
+      List< String> strings = Arrays.asList("foo", "bar");
+      System.out.print(strings.get(0));
+    }
+
+    public boolean badWhiteSpaces()
+    {
+        int sum = 2+2;
+        int diff = 2-1;
+        System.out. println("hello");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '.' is followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.cloudmok.org/coding_rules#rule_key=checkstyle%3ANo_Whitespace_After___Custom)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The String "hello" appears 2 times in the file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.cloudmok.org/coding_rules#rule_key=checkstyle%3AMultiple_String_Literals_Custom)